File: samples/sample.pl
#!/usr/bin/perl
# Perl script: foobar.pl

use strict;
use warnings;

# Defining variables
my $foo = "Lorem Ipsum";
my $bar = 123;

# Displaying variables
print "Foo: $foo\n";
print "Bar: $bar\n";

# Using an array
my @baz = (1, 2, 3, 4, 5);
print "@baz\n";

# Using a hash
my %qux = ("foo" => "bar", "baz" => "qux");
while (my ($key, $value) = each %qux) {
    print "Key: $key, Value: $value\n";
}

# Using control structures
if (exists $qux{"foo"}) {
    print "Found foo: $qux{'foo'}\n";
} else {
    print "Did not find foo\n";
}

# Using a switch-like statement
if ($bar == 123) {
    print "Bar is 123\n";
} else {
    print "Bar is not 123\n";
}

# Using a loop
foreach my $i (@baz) {
    print "$i\n";
}

# Using a subroutine
sub double {
    my ($x) = @_;
    return 2 * $x;
}

print "Double of Bar: ", double($bar), "\n";

File: samples/sample.jl
# Julia script: foobar.jl

# Defining variables
foo = "Lorem Ipsum"
bar = 123

# Displaying variables
println("Foo: ", foo)
println("Bar: ", bar)

# Using an array
baz = [1, 2, 3, 4, 5]
println(baz)

# Using a dictionary
qux = Dict("foo" => "bar", "baz" => "qux")
println(qux)

# Using control structures
if haskey(qux, "foo")
    println("Found foo: ", qux["foo"])
else
    println("Did not find foo")
end

# Using a switch-like statement
bar == 123 ? println("Bar is 123") : println("Bar is not 123")

# Using a loop
for i in 1:5
    println(i)
end

# Using a function
function double(x)
    return 2 * x
end

println("Double of Bar: ", double(bar))
